node-notes-cli/app.js

```diff
@@ -0,0 +1,51 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const data = JSON.parse(await readFile('data.json'));
+const notes = data.notes;
+
+async function dataFile() {
```

Comment:
I like that you encapsulate the write. I think the name of this function is awkward, though. Since functions "do things", it's usually best to name functions starting with a verb, like `writeDataFile` or similar.